samza-core/src/main/java/org/apache/samza/container/host/PosixCommandBasedStatisticsGetter.java

```diff
@@ -85,4 +86,10 @@ public SystemMemoryStatistics getSystemMemoryStatistics() {
       return null;
     }
   }
+
+  @Override
+  public ProcessCPUStatistics getProcessCPUStatistics() {
+    throw new NotImplementedException(
+        "No appropriate Posix command available for getting recent CPU usage information. For example, the CPU information exposed by ps command 'ps -o %cpu= -p <PID>' represents the percentage of time spent running during the entire lifetime of a process not for the recent CPU usage");
+  }
```

Comment:
Why not `top`?

Reply:
Yes, `top` can give the instant/recent CPU usage information for a given process, but it has below cons so that I didn't choose it:

- Need to have operating system specific implementation, since the top command is not a POSIX command and its input and output format in Linux and macOS are different.
- Slower than ps command. Especially, on macOS the top command needs to get at least two samples to get the right CPU usage, which takes longer time(~ 2 seconds) than 1 sample.
  - not clear the reason, but it shows the consistent behavior that one sample(-l1) always return 0.0 on my macOS